src/commands/src/shared/next-gen-stage.ts

```diff
@@ -1,46 +1,74 @@
-import { executeCliCommand } from '../helpers';
+import { Job } from '../../../entities/job';
+import { CommandExecutorResponse, CommandExecutorResponseStatus } from '../../../services/commandExecutor';
+import { executeCliCommand, getRepoDir } from '../helpers';
 
 const DOCS_WORKER_USER = 'docsworker-xlarge';
 interface StageParams {
-  repoDir: string;
-  mutPrefix: string;
-  projectName: string;
-  bucket: string;
-  url: string;
-  patchId?: string;
-  commitBranch: string;
-  commitHash: string;
+  job: Job;
+  bucket?: string;
+  url?: string;
 }
 
-export async function nextGenStage({
-  mutPrefix,
-  projectName,
-  bucket,
-  url,
-  patchId,
-  commitBranch,
-  commitHash,
-}: StageParams) {
-  let hostedAtUrl = `${url}/${mutPrefix}/${DOCS_WORKER_USER}/${commitBranch}/`;
-  let prefix = mutPrefix;
+export async function nextGenStage({ job, bucket, url }: StageParams): Promise<CommandExecutorResponse> {
+  const { mutPrefix, branchName, patch, project, newHead } = job.payload;
```

Comment:
Just like in the [`next-gen-parse.ts` file](https://github.com/mongodb/docs-worker-pool/pull/938#discussion_r1431646872), we will want to use the `patchId` value that we get when we call the `prepareBuildAndGetDependencies` function as opposed to the `patch` property from `job.payload`.

Reply:
Great call out!